src/psij/executors/batch/batch_scheduler_executor.py

```diff
@@ -0,0 +1,539 @@
+import logging
+import subprocess
+import time
+import traceback
+from abc import abstractmethod
+from pathlib import Path
+from threading import Thread, RLock
+from typing import Optional, List, Dict, Collection, cast, TextIO
+
+from psij import JobExecutor, JobExecutorConfig, Launcher, Job, SubmitException, \
+    JobStatus, JobState
+from psij.executors.batch.template_function_library import ALL as FUNCTION_LIBRARY
+
+
+logger = logging.getLogger(__name__)
+
+
+class BatchSchedulerExecutorConfig(JobExecutorConfig):
+    """A base configuration class for :class:`~BatchSchedulerExecutor` implementations.
+
+    When subclassing :class:`~BatchSchedulerExecutor`, specific configuration classes inheriting
+    from this class should be defined, even if empty.
+    """
+
+    def __init__(self, launcher_log_file: Optional[Path] = None,
+                 work_directory: Optional[Path] = None, queue_polling_interval: int = 30,
+                 queue_polling_error_threshold: int = 2,
+                 keep_files: bool = False):
+        """Initializes a base batch scheduler executor configuration.
+
+        Parameters
+        ----------
+        launcher_log_file
+            See :func:`~JobExecutorConfig.__init__`.
+        work_directory
+            A directory where submit scripts and auxiliary job files will be generated.
```

Comment:
I should probably make a note that this should be on a shared FS, since the exit code file is written on the rank 0 node and read on the login node. And the system should be fail-fast if the workdir is not on a shared fs.

Reply:
Fixed